softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java

```diff
@@ -17,16 +17,24 @@
 
     @Override
     public Response<R> handle(Request<?> msg) {
-        Request<T> tRequest = new Request<>();
-        tRequest.setHeader(msg.getHeader());
-        JacksonDataMapper jacksonDataMapper = new JacksonDataMapper();
-
-        Class thisClass = AbstractRequestHandler.class;
-        ParameterizedType type = (ParameterizedType) thisClass.getGenericSuperclass();
-        Class parameter = (Class) type.getActualTypeArguments()[0];
-        tRequest.setData(jacksonDataMapper.convert((Map<String, Object>)msg.getData(), (Class<T>) parameter));
-
-        return doHandle(tRequest);
+//        Request<T> tRequest = new Request<>();
+//        tRequest.setHeader(msg.getHeader());
+//        JacksonDataMapper jacksonDataMapper = new JacksonDataMapper();
+//
+//        Class thisClass = AbstractRequestHandler.class;
+//        ParameterizedType type = (ParameterizedType) thisClass.getGenericSuperclass();
+//        Class parameter = (Class) type.getActualTypeArguments()[0];
+//        tRequest.setData(jacksonDataMapper.convert((Map<String, Object>)msg.getData(), (Class<T>) parameter));
+
+
+        DataMapper mapper = new JacksonDataMapper();
+        Map<String, Object> map = mapper.convertToMap(msg);
+        RequestData data = mapper.convert(map, RequestData.class);
```

Comment:
@VadymPolyanski Why you need two conversions? What will you get when trying to convert to interface type ?